### PR TITLE
Grabbers can now only contain things smaller than the assembly.

### PR DESCRIPTION
**Pull request by DaxDupont**
[Changelogs]: 

:cl: Dax Dupont
balance: Grabbers/throwers no longer can contain/throw things equal to the assembly size. 
/:cl:

[why]: I think this was a logic error.
Grabbers can hold 10 items, so arguably the objects shouldn't be the same weight class as the assembly holding the object.

### CODE REVIEW

**Comment by arsserpentarium** [comment]
ok

**Comment by DaxDupont** [comment]
actually never mind they should be allowed to throw things that are larger.

**Comment by evsey9** [comment]
Aaaactuaally I repeatedly stated in my PR that this WAS NOT a logic issue.

**Comment by evsey9** [comment]
Also this ruins all small circuit bots that use grabbers by making them completely useless

**Comment by arsserpentarium** [comment]
At first autothrowing of assemblies was a bug. But I found it hilarious and leaved.
If you do this, barely you will make some deathmachine harder to make. But will remove fun.

**Comment by DaxDupont** [comment]
Small circuits should not be able to hold 10 small items

**Comment by DaxDupont** [comment]
They can still hold tiny items

**Comment by arsserpentarium** [comment]
I am not about grabbers. I think,if grabbers will store only smaller things,it's ok.
I was talking about throwers

**Comment by DaxDupont** [comment]
@evesy
Hatchets and this are related. 
Small circuits can still store tiny items. 

The problem was that circuits could store 10 hatchets and throw them all at once.
We should not have some magical storage that can hold 10 same weight items.

Circuits aren't hard to get when you literally copy paste the code from the forums or reddit.

I'll concede that this is balance because originally I didn't expect the original coder to be retarded

**Comment by DaxDupont** [comment]
If you want to add bluespess grabbers that's fine, as long they hold equal weight or whatever.

I'd make that a separate thing so grabbers don't become insanely expensive for other purposes.

**Comment by DaxDupont** [comment]
Also I'd like to add I did this before I saw you the changelog of your PR.

**Comment by Tlaltecuhtli** [comment]
why does it have a fix tag

**Comment by DaxDupont** [comment]
because it doesn't update automatically

**Comment by ACCount12** [comment]
That's not a fix, that's an "I DED"-tier balance meme. @arsserpentarium wrote both thrower and grabber, and he says it was intentional. I rewrote both thrower and grabber, and I say this was intentional.

Grabber may need some nerfs, but it's hard to abuse it, so I'd say it's fine as it is.

**Comment by DaxDupont** [comment]
I changed the tag in the changelog, please read. Also it's easy to abuse
when there's a ready made circuit on the forums

**Comment by arsserpentarium** [comment]
Still tag "i ded" is required.

**Comment by DaxDupont** [comment]
I don't mess with circuits much but when I saw people running around with really weird things and then I see the forum post that says it instacrit people I went to find out what the fuck is wrong with it.

**Comment by evsey9** [comment]
perhaps its throwers that needs a nerf and not grabbers?
you can still instacrit people even with this PR merged lmao

**Comment by DaxDupont** [comment]
Yeah with bigger bots.

Though that's an issue on it's own. Circuits don't become bags of holding for some reason.
I am 100% okay with there being a variant that consumes bluespess mesh.

Also with the throwers, I'd like some input on how to fix/balance them. Considering the issue lies with them firing at the same time, and not the delay as I previously thought.

**Comment by evsey9** [comment]
circuits don't become bags of holding, they don't go onto your belt, backpack or pocket slots ( the bigger ones )
also normal sized circuits still can do that as far as i know

**Comment by DaxDupont** [comment]
They can fit small or tiny yes,
Bulky can fit normal/small/tiny.

Like I said, I am still looking for something that prevents throwers from causing issues mass throwing, but I am not sure how to do that without the raising the complexity but that's not a good option.

**Comment by DaxDupont** [comment]
@arsserpentarium I would like your input on how to deal with throwers being able to toss all of the contents at once.

**Comment by arsserpentarium** [comment]
ideas:
1.cooldown 0.1 s for extenal manipulation. Or cap on external activity per tick. So you just can't fire them all at once.
2.make circuit require capacitor. Which limits power of assembly.So 10 throwers at once hits power cap.
3.limit slots or external manipulation;make these slots to be "ports" on circuit case sprite.And for every manipulative circuit make their appearance from that port.
Would be cool, because provides custom sprites as well, but too complex to make, maintain and play.

File: code/modules/integrated_electronics/subtypes/manipulation.dm [hunk 1]
```diff
@@ -315,16 +315,10 @@
 		if(mode == 1)
 			if(check_target(AM))
 				var/weightcheck = FALSE
-				if ((!istype(AM,/obj/item/device/electronic_assembly/)) && (!istype(AM,/obj/item/device/transfer_valve)))
```
Comment (ShizCalev) [inline code/modules/integrated_electronics/subtypes/manipulation.dm]: should probably keep the transfer valve check

**Pull request merged**